src/sentry/sentry_metrics/consumers/indexer/batch.py

```diff
@@ -431,7 +432,8 @@ def reconstruct_messages(
                     "timestamp": old_payload_value["timestamp"],
                     "project_id": old_payload_value["project_id"],
                     "type": old_payload_value["type"],
-                    "value": old_payload_value["value"],
+                    # Metrics doesn't support gauges, so skip type check here
+                    "value": old_payload_value["value"],  # type: ignore
```

Comment:
`# type: ignore` should only be used for bugs in the type checker -- please fix the actual problem -- probably by adjusting the types that are being assigned or the TypedDict